myapp/utility.py

```diff
@@ -1,116 +1,880 @@
-from flask import make_response
-import db
-import schema
+from functools import wraps
+from flask import make_response, render_template, jsonify, redirect, url_for
+from flask.ext.login import current_user
+from myapp import models as m
+from myapp import (app, db)
+import linecache
+import sys
+import re
+ds = db.session
 
 
-# Utility functions
-# Generate a lab's id according to lab_name, class_name and prof_name
-def Generate_lab_id(lab_name,class_name,prof_name):
+# --- Debug/Logging functions ---
+# make things easier to spot in terminal by printing them out
+def cprint(*args):
```

Comment:
once again, please look into django logging. this function, and pretty much anything using print, should never make it into production code, unless you are writing a logger.